packages/datx/src/helpers/model/fields.ts

```diff
@@ -41,6 +41,14 @@ export function updateRef(
 ): PureModel | Array<PureModel> | null {
   const bucket: IBucket<PureModel> | undefined = getMeta(model, `ref_${key}`);
 
+  if (isIdentifier(value) || Array.isArray(value)) {
+    const fieldDef = getMeta(model, MetaModelField.Fields, {})[key];
+    const type = fieldDef.referenceDef.model;
+    value = mapItems(value, (v: IIdentifier | IModelRef | PureModel) =>
+      isIdentifier(v) ? { id: v, type } : getModelRef(v),
```

Comment:
This won't work, because the model from fieldDef can also be a function: https://github.com/infinum/datx/blob/release/v2/packages/datx/src/Attribute.ts#L37

The best thing to do here would be to pass the type trough the `getModelRefType ` method https://github.com/infinum/datx/blob/release/v2/packages/datx/src/helpers/model/init.ts#L29